### PR TITLE
Getting up to date with sdk-ts

### PR DESCRIPTION
**Pull request by krider2010**
This PR implements TypeScript SDK feature parity in the Go SDK, adding testing  infrastructure and advanced event processing capabilities.

### Testing Infrastructure

  - Ginkgo/Gomega testing framework with BDD-style test organization
  - Comprehensive test helpers including HTTP mock server infrastructure
  - Configurable mock clients for reliable testing without network dependencies
  - Complete test coverage for all new features with edge case handling

### Event Metadata Support

  - Metadata fields added to Event and EventResponse structs
  - Backward compatibility maintained with optional metadata handling
  - Publishing support via WithMetaData() option for event publishing
  - Example integration showing metadata usage patterns

### Advanced Subscription Filters

  - Complex filter types supporting multiple conditions with logical operators
  - Rate limiting and deduplication options for push subscriptions
  - Backward compatibility with existing filter patterns via LegacyFilter support
  - Type-safe filter definitions with proper JSON marshalling

### HMAC Push Subscription Verification

  - HMAC-SHA256 signature verification for webhook security
  - Timing-safe comparison to prevent timing attacks
  - Configurable tolerance for timestamp validation with sensible defaults
  - Comprehensive error types with specific error codes for debugging
  - Complete verification workflow with header parsing and validation

### Long-Running Subscribers

  - Concurrent processing with configurable processors per subscription
  - Graceful lifecycle management with context-based cancellation
  - Automatic retry logic with exponential backoff and configurable limits
  - Error handling framework with optional error callbacks and monitoring
  - Multiple subscription support running concurrently in separate goroutines
  - Production-ready patterns with proper resource cleanup and shutdown handling

### Build Tools

  - Simplified taskfile.yml with essential test and build commands
  - Ginkgo test runner integration for enhanced test output
  - Example applications demonstrating real-world usage patterns

### CODE REVIEW

File: examples/long_running_subscriber/main.go [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+	// Graceful shutdown
+	fmt.Println("Stopping subscriber...")
+	subscriber.Stop()
+	fmt.Println("Subscriber stopped.")
+	fmt.Printf("Final stats: %s\n", stats)
+}
+
+func registerHandlers(subscriber *sailhouse.SailhouseSubscriber, stats *ProcessingStats) {
+	// Handler for user events
+	subscriber.Subscribe("user-events", "user-processor", func(ctx context.Context, event *sailhouse.Event) error {
```
Comment (estephinson) [inline examples/long_running_subscriber/main.go:106]: Each event type should be it's own topic imo?

File: sailhouse/subscriber.go [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+	s.ctx, s.cancel = context.WithCancel(ctx)
+	s.running = true
+
+	// Start processors for each subscription
+	for _, subscription := range s.subscriptions {
+		for i := 0; i < s.options.PerSubscriptionProcessors; i++ {
+			s.wg.Add(1)
+			go s.runProcessor(subscription, i)
+		}
+	}
```
[the remark this answers is not in the document]
Comment (krider2010) [reply sailhouse/subscriber.go:123]: I was going to say that I'm no expert in coroutines or similar so wanted you to triple check that

File: examples/long_running_subscriber/main.go [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+Production considerations:
+- Add proper logging with structured logs
+- Implement proper error tracking (e.g., Sentry)
+- Add metrics collection (e.g., Prometheus)
+- Configure appropriate retry counts and delays
+- Implement circuit breakers for external service calls
+- Add health check endpoints
+- Use proper configuration management
+- Implement dead letter queue handling for failed events
+*/
```
[the remark this answers is not in the document]
Comment (krider2010) [reply examples/long_running_subscriber/main.go:364]: I got AI to do all my examples!

[the remark this answers is not in the document]
Comment (krider2010) [reply examples/long_running_subscriber/main.go:364]: Yeah will fix

File: examples/long_running_subscriber/main.go (hunk 1 above)
Comment (krider2010) [reply examples/long_running_subscriber/main.go:106]: AI... Agreed

File: examples/long_running_subscriber/main.go [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+		case "user.signup":
+			return handleUserSignup(eventData, event.Metadata)
+		case "user.email_verified":
+			return handleEmailVerified(eventData, event.Metadata)
+		case "user.trial_started":
+			return handleTrialStarted(eventData, event.Metadata)
+		default:
+			fmt.Printf("[REGISTRATION] Unknown event type: %s\n", eventType)
+			return nil
+		}
```
Comment (estephinson) [inline examples/long_running_subscriber/main.go:134]: I think this is in all of the examples here - worth revisiting I think (or maybe just not adding for now)

Comment (krider2010) [reply examples/long_running_subscriber/main.go]: Revisted

**Review by estephinson — APPROVED** [review]
